apps/src/storage/dataBrowser/dataVisualizer/Snapshot.jsx

```diff
@@ -0,0 +1,122 @@
+/* global ClipboardItem */
+import React from 'react';
+import PropTypes from 'prop-types';
+import {connect} from 'react-redux';
+import moment from 'moment';
+import msg from '@cdo/locale';
+import {html2canvas} from '@cdo/apps/util/htmlToCanvasWrapper';
+import BaseDialog from '@cdo/apps/templates/BaseDialog';
+import PendingButton from '@cdo/apps/templates/PendingButton';
+import * as dataStyles from '../dataStyles';
+
+class Snapshot extends React.Component {
+  static propTypes = {
+    chartTitle: PropTypes.string.isRequired,
+    selectedOptions: PropTypes.string.isRequired,
+    // Provided via Redux
+    tableName: PropTypes.string.isRequired,
+    projectName: PropTypes.string.isRequired
+  };
+
+  state = {
+    isSnapshotOpen: false,
+    isCopyPending: false,
+    isSavePending: false
+  };
+
+  handleOpen = () => this.setState({isSnapshotOpen: true});
+  handleClose = () => this.setState({isSnapshotOpen: false});
+
+  getPngBlob = () => {
+    const element = document.getElementById('snapshot');
+    const options = {
+      background: '#fff'
+    };
+    return html2canvas(element, options).then(canvas => {
+      return new Promise(function(resolve, reject) {
+        canvas.toBlob(function(blob) {
+          resolve(blob);
+        });
+      });
+    });
+  };
+
+  copy = () => {
+    this.setState({isCopyPending: true});
+    this.getPngBlob().then(pngBlob => {
+      try {
+        navigator.clipboard.write([
+          new ClipboardItem({
+            'image/png': pngBlob
+          })
+        ]);
+        this.setState({isCopyPending: false});
+      } catch (error) {
+        console.error(error);
+      }
+    });
+  };
```

Comment:
I've got three thoughts on this copy function.

1. [`clipboard.write`](https://developer.mozilla.org/en-US/docs/Web/API/Clipboard/write) is also an asynchronous operation, but we're not treating it as such here.
2. Is it intentional that we're not resetting the copy pending state if an error occurs?
3. This seems like a great place to use async/await!

Reply:
Done!